java/test/src/main/java/io/ray/test/MetricTest.java

```diff
@@ -74,14 +123,136 @@ public void testAddHistogram() {
     boundaries.add(15.0);
     boundaries.add(12.0);
     Histogram histogram = new Histogram("metric_histogram", "histogram", "1pc",
-        boundaries, tags);
+      boundaries, tags);
     for (int i = 1; i <= 200; ++i) {
       histogram.update(i * 1.0d);
-      histogram.record();
     }
+    Assert.assertTrue(doubleEqual(200.0d, histogram.getValue()));
     List<Double> window = histogram.getHistogramWindow();
     for (int i = 0; i < Histogram.HISTOGRAM_WINDOW_SIZE; ++i) {
       Assert.assertTrue(doubleEqual(i + 101.0d, window.get(i)));
     }
+    histogram.record();
+    Assert.assertTrue(doubleEqual(200.0d, histogram.getValue()));
+    Assert.assertEquals(window.size(), 0);
+  }
+
+  @Test
+  public void testRegisterGauge() throws InterruptedException {
+    TestUtils.skipTestUnderSingleProcess();
+    Gauge gauge = registerGauge();
+
+    gauge.update(2.0);
+    Assert.assertTrue(doubleEqual(gauge.getValue(), 2.0));
+    gauge.update(5.0);
+    Assert.assertTrue(doubleEqual(gauge.getValue(), 5.0));
+    RayMetrics.shutdown();
   }
+
+  @Test
+  public void testRegisterCount() throws InterruptedException {
+    TestUtils.skipTestUnderSingleProcess();
+    Count count = registerCount();
+
+    count.inc(10.0);
+    count.inc(20.0);
+    Assert.assertTrue(doubleEqual(count.getCount(), 30.0));
+    count.inc(1.0);
+    count.inc(2.0);
+    Assert.assertTrue(doubleEqual(count.getCount(), 33.0));
+    RayMetrics.shutdown();
+  }
+
+  @Test
+  public void testRegisterSum() throws InterruptedException {
+    TestUtils.skipTestUnderSingleProcess();
+    Sum sum = registerSum();
+
+    sum.update(10.0);
+    sum.update(20.0);
+    Assert.assertTrue(doubleEqual(sum.getSum(), 30.0));
+    sum.update(1.0);
+    sum.update(2.0);
+    Assert.assertTrue(doubleEqual(sum.getSum(), 33.0));
+    RayMetrics.shutdown();
```

Comment:
We can use `@AfterMethod` to avoid duplicating this line.

Reply:
OK.